typed-store/src/rocks/mod.rs

```diff
@@ -21,6 +21,37 @@ mod tests;
 type DBRawIteratorMultiThreaded<'a> =
     rocksdb::DBRawIteratorWithThreadMode<'a, DBWithThreadMode<MultiThreaded>>;
 
+#[macro_export]
+// It generates a method (reopencfs) to reopen a list of column families.
+// The macro should be called by passing in the column family names and the
+// corresponding DBMap Key-Value types.
+//
+// Example: "cf_1";<String, u32>, "cf_2";<u32, u32>
+macro_rules! reopen {
+    ( $($cf:expr;<$K:ty, $V:ty>),* ) => {
```

Comment:
If you were to do this:
```rust
    ( $db_ref: expr; $($cf:expr;<$K:ty, $V:ty>),* ) => {
```
 would it allow you directly write the returned tuple a couple of lines later?
 ```rust 
$(
    DBMap::<$K, $V>::reopen($db_ref, Some($cf)).expect(&format!("Cannot open {} CF.", $cf)[..])
),
```

Hence the call site would look like:
```rust
let (db_cf_1 db_cf_2) =   reopen!(&rocks FIRST_CF;<i32, String>, SECOND_CF;<i32, String>);
```

Advantage: I don't have to remember that `reopencfs` exists it's all in 1 place.
Disadvantage: i don't get as fancy a type error (though failure to expand is still a static error).

Reply:
Ah, to be honest that was my original intention but yesterday when I tried it to do it like this, I wasn't able to successfully pass a reference (db) to use together with the repetitive input, that's why I went with the `reopencfs` approach. Tried again today using your recommendation and it works like a charm :-) - I can't even remember what I was doing wrong. Thanks a lot! Now there is no need to call a separate auto-generated method.